plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -185,10 +189,16 @@ public GlueHiveMetastore(
         this.defaultDir = glueConfig.getDefaultWarehouseDir();
         this.catalogId = glueConfig.getCatalogId().orElse(null);
         this.partitionSegments = glueConfig.getPartitionSegments();
-        this.executor = requireNonNull(executor, "executor is null");
-        this.columnStatisticsProvider = requireNonNull(columnStatisticsProvider, "columnStatisticsProvider is null");
+        this.partitionsReadExecutor = requireNonNull(partitionsReadExecutor, "executor is null");
         this.assumeCanonicalPartitionKeys = glueConfig.isAssumeCanonicalPartitionKeys();
         this.tableFilter = requireNonNull(tableFilter, "tableFilter is null");
+        this.enableColumnStatistics = hiveConfig.isTableStatisticsEnabled();
+        if (this.enableColumnStatistics) {
```

Comment:
please setup that in the module:
```java
        installModuleIf(
                HiveConfig.class,
                HiveConfig::isTableStatisticsEnabled,
                innerBinder -> innerBinder.bind(GlueColumnStatisticsProvider.class).to(DefaultGlueColumnStatisticsProvider.class).in(Scopes.SINGLETON),
                innerBinder -> innerBinder.bind(GlueColumnStatisticsProvider.class).to(DisabledGlueColumnStatisticsProvider.class).in(Scopes.SINGLETON));

```
should do

Reply:
Oh I see that it would require some more refactor regarding stats so those are guice bound too. I guess it can be done as a followup.